script/generate-documentation

```diff
@@ -15,20 +15,47 @@
 # You should have received a copy of the GNU General Public License along
 # with this program; if not, see <http://www.gnu.org/licenses/>.
 
-# the script accepts input arguments:
+# The script runs in three major modes:
```

Comment:
I really like that you wrote all this documentation.

Reply:
Actually I wrote it the moment I understand more or less defined way how it should be used since we added 'pull request mode' (because we don't give CI write permissions to upstream anymore)